.1xx sdk feature band extended support policy update.

ashnaga opened the pull request:
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)


<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/porting/versioning-sdk-msbuild-vs.md](https://github.com/dotnet/docs/blob/e71af802804cb89eb01a48fb1646f2a955cd74c2/docs/core/porting/versioning-sdk-msbuild-vs.md) | [Overview of .NET, MSBuild, and Visual Studio versioning](https://review.learn.microsoft.com/en-us/dotnet/core/porting/versioning-sdk-msbuild-vs?branch=pr-en-us-35211) |


<!-- PREVIEW-TABLE-END -->

ashnaga committed:
.1xx sdk feature band extended support policy update.

adegeo commented on docs/core/porting/versioning-sdk-msbuild-vs.md:
```diff
@@ -51,6 +51,8 @@ The support timeframe for the SDK typically matches that of the Visual Studio ve
 >
 > <sup>5</sup> 7.0.200 and newer SDKs require a minimum Visual Studio version of 17.4.
+> 
+> <sup>6</sup> Starting .NET 6, .1xx SDK feature band is supported throughout the lifecycle of .NET major versions. During the extended period, support is limited to security fixes and minimal absolutely required fixes (non-security) for Linux only. To learn more about the reasoning for this extended support, refer to this [document](https://github.com/dotnet/source-build#support).
```
Some minor clean up here. I would also change the "document" link text to something more descriptive.

```suggestion
> <sup>6</sup> Starting with .NET 6, .1xx SDK feature band is supported throughout the lifecycle of major .NET versions. During the extended support period, support is limited to security fixes and minimal high-priority non-security fixes for Linux only. To learn more about the reasoning for this extended support, refer to this [document](https://github.com/dotnet/source-build#support).
```

adegeo reviewed (changes requested):
One thing I'm confused about is that you did `<sup>6</sup>` to add a note `#6`, but where is `#6` referenced in the article?

ashnaga committed:
Update docs/core/porting/versioning-sdk-msbuild-vs.md

Co-authored-by: Andy (Steve) De George <67293991+adegeo@users.noreply.github.com>

ashnaga commented:
Tagging Michael and Marc to review. Cannot add the reviewer and so tagging in here.
@MichaelSimons @marcpopMSFT

MichaelSimons commented on docs/core/porting/versioning-sdk-msbuild-vs.md:
```diff
@@ -51,6 +51,8 @@ The support timeframe for the SDK typically matches that of the Visual Studio ve
 >
 > <sup>5</sup> 7.0.200 and newer SDKs require a minimum Visual Studio version of 17.4.
+> 
+> <sup>6</sup> Starting with .NET 6, .1xx SDK feature band is supported throughout the lifecycle of major .NET versions. During the extended support period, support is limited to security fixes and minimal high-priority non-security fixes for Linux only. To learn more about the reasoning for this extended support, refer to this [document](https://github.com/dotnet/source-build#support).
```
It doesn't feel necessary to say `Starting with .NET 6` because as of now all in-support versions have this support policy.  Over time it will become meaningless to start with this.

ashnaga committed:
Removed - "starting .NET 6"

ashnaga committed:
Moving out of <sup> as applicable for any .NET version and not related to VS.

ashnaga commented:
> One thing I'm confused about is that you did `<sup>6</sup>` to add a note `#6`, but where is `#6` referenced in the article?

Addressed the comment

adegeo commented on docs/core/porting/versioning-sdk-msbuild-vs.md:
```diff
@@ -41,6 +41,8 @@ The support timeframe for the SDK typically matches that of the Visual Studio ve
 > [!NOTE]
 > Targeting `net6.0` is officially supported in Visual Studio 17.0+ only.
 > Targeting `net7.0` is officially supported in Visual Studio 17.4+ only.
+> 
```
```suggestion
>
```

adegeo committed:
Apply suggestions from code review
